arelle/plugin/validate/NL/rules/fg_nl.py

```diff
@@ -193,3 +195,44 @@ def rule_fg_nl_09(
             msg=_('A fact SHOULD NOT contain an "id" attribute'),
             modelObject=facts,
         )
+
+
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18,
+    ],
+)
+def rule_fg_nl_11(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FG-NL-11: XBRL instance file Should only contain standard units based on the XBRL units Registry.
+    """
+    modelXbrl = val.modelXbrl
+    customUnits = []
+    utrItemTypeEntries = ValidateUtr(modelXbrl).utrItemTypeEntries
+    validUtrUnits = set()
+    for entries in utrItemTypeEntries.values():
+        for entry in entries.values():
+            if entry.nsUnit and entry.unitId:
+                #unitQname = entry.nsUnit.rsplit("/")[-1] + ':' + entry.unitId
+                validUtrUnits.add((entry.nsUnit, entry.unitId))
+    for unitId, unit in modelXbrl.units.items():
+        for measures in unit.measures:
+            for measure in measures:
+                if (measure.namespaceURI, measure.localName) not in validUtrUnits:
+                    customUnits.append(measure)
+    if customUnits:
+        for unit in customUnits:
+            yield Validation.warning(
+                codes='NL.FG-NL-11',
+                msg=_('Standard units SHOULD be used. "%(modelUnit)s" is not a standard unit.'),
+                modelObject=unit,
```

Comment:
Since you're adding measures (QNames) to `customUnits` above, `unit` here is actually a measure QName, not a ModelObject/ModelUnit. I'd suggest having `customUnits` being a list of tuples of `(unit: ModelUnit, measure: QName)` so you can then pass `unit` as `modelObject` and `measure` into the message itself.